pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -504,21 +504,15 @@ func (c *Controller) getEgressServiceAddrs() ([]model.NetworkGateway, []string)
 	var addrs []model.NetworkGateway
 	var sas []string
 	for _, subset := range endpoints.Subsets {
-		for index, address := range subset.Addresses {
-			if subset.Ports[index].Port == common.DefaultFederationPort {
-				ips, err := c.getIPAddrsForHostOrIP(address.IP)
-				if err != nil {
-					c.logger.Errorf("error converting to IP address from %s: %s", address.IP, err)
-					continue
-				}
-				for _, ip := range ips {
-					addrs = append(addrs, model.NetworkGateway{
-						Addr: ip,
-						Port: uint32(common.DefaultFederationPort),
-					})
-					sas = append(sas, serviceAccountByIP[ip])
```

Comment:
Shouldn't we also check whether `serviceAccountByIP` has a value at given ip? I guess that we cannot be 100% sure that ServiceAccounts and Endpoints are in sync. Then `sas` might contain empty strings. Do you agree with me?

Reply:
Yup, looks like `sas` can contain multiple empty strings and the same service account name repeated several times. While neither empty strings or repeated SA names break anything, we should clean this up and make `sas` a set.

Reply:
Thanks. I fixed it, look at the last commit.